lib/wab/impl/data.rb

```diff
@@ -83,73 +71,51 @@ def set(path, value, repair=false)
         else
           validate_value(value)
         end
-        node = @root
         path = path.to_s.split('.') unless path.is_a?(Array)
-        path[0..-2].each { |key|
-          if node.is_a?(Hash)
-            key = key.to_sym
-            node[key] = {} unless node.has_key?(key)
-            node = node[key]
-          elsif node.is_a?(Array)
-            key = key_to_int(key)
-            if key < node.length && -node.length < key
-              node = node[key]
-            else
-              nn = {}
-              if key < -node.length
-                node.unshift(nn)
-              else
-                node[key] = nn
-              end
-              node = nn
-            end
-          else
-            raise StandardError.new("Can not set a member of an #{node.class}.")
-          end
-        }
+        node = assign_node(path, root)
+
         key = path[-1]
+        ensure_hash_or_array(node)
```

Comment:
Raising an error situation so need for performance there. Checking is_a? Array or Hash twice puts that check on every call.